core-bundle/src/Intl/Countries.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Intl;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\System;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Intl\Countries as SymfonyCountries;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class Countries
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $contaoFramework;
+
+    /**
+     * @var array<string>
+     */
+    private $countriesList;
+
+    public function __construct(TranslatorInterface $translator, RequestStack $requestStack, ContaoFramework $contaoFramework, array $countriesList)
+    {
+        $this->translator = $translator;
+        $this->requestStack = $requestStack;
+        $this->contaoFramework = $contaoFramework;
+        $this->countriesList = $countriesList;
+    }
+
+    /**
+     * @return array<string,string> Translated country names indexed by their uppercase ISO 3166-1 alpha-2 code
+     */
+    public function getCountries(string $displayLocale = null): array
+    {
+        if (null === $displayLocale && ($request = $this->requestStack->getCurrentRequest())) {
+            $displayLocale = $request->getLocale();
+        }
+
+        $countries = SymfonyCountries::getNames($displayLocale);
+        $needsResort = false;
+
+        if (\count($this->countriesList)) {
```

Comment:
Personally, I'd go for either…
```suggestion
        if (\count($this->countriesList) > 0) {
```
…or…
```suggestion
        if (!empty($this->countriesList)) {
```
instead, because it IMHO lowers the cognitive load to understand the code.

Reply:
I always forget what the coding style is here.

Personally I’d favor `if ($this->countriesList) {`

@leofeyer ?

Reply:
Changed in 608d942ae7fe3a25afc3a8afcf7e3645f9ed25f2